code/modules/spells/spell_types/shadow_walk.dm

```diff
@@ -78,3 +82,29 @@
 			visible_message("<span class='boldwarning'>[jaunter] emerges from the darkness!</span>")
 		playsound(loc, 'sound/magic/ethereal_exit.ogg', 50, TRUE, -1)
 	qdel(src)
+
+/obj/effect/dummy/phased_mob/shadow/Initialize(mapload)
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/effect/dummy/phased_mob/shadow/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	. = ..()
+
+/obj/effect/dummy/phased_mob/shadow/process()
+	if(!jaunter)
+		qdel(src)
+	if(jaunter.loc != src)
+		qdel(src)
+	check_light_level()
+	shadow_heal()
+
+/obj/effect/dummy/phased_mob/shadow/ex_act()
+	return
+
+/obj/effect/dummy/phased_mob/shadow/bullet_act()
+	return BULLET_ACT_FORCE_PIERCE
+
+/obj/effect/dummy/phased_mob/shadow/singularity_act()
+	return
```

Comment:
1.The phased mob is already processing a few lines up
2.Are these overrides of acts() needed? why are they here?
3.Are you sure this is unintended behavior? If you end up rebalancing the healing back up to the old rates are you even changing anything here? 
4.Is this a fix or balance pr?

Reply:
ok, everything in that segment is messed up somehow. I didn't touch any of that code but it somehow got put in
3. Its odd behavior that being in darkness heals you slowly but being in darkness "while moving" heals you very fast
4. well you see nightmares are used to being able to heal quickly by moving fast. making them heal slow all of the time seems like a jump that would change how nightmares are played.
Ill fix the shit

Reply:
this isn't even my code 🦈